### ISSUE TITLE
Enable hevm in ds-test tests, when it is possible

### ISSUE THREAD

**Issue by travs** [melonproject/protocol#194]
Skipping `ds-test` using `hevm` for now, since it does not yet support libraries, so almost all of our tests fail. The tests work now just using `ethrun`  instead.

Just remove the `export  DAPP_NO_HEVM=1` [introduced here](https://github.com/melonproject/protocol/commit/93cb6b534fdc185b21af33bf1a5a06f04e2f762f#diff-b9cfc7f2cdf78a7f4b91a753d10865a2R11) when this is closed: https://github.com/dapphub/hevm/issues/29

**Comment by mbrock** [comment]
The issue dapphub/hevm#29 is now closed, so you could try if it works for you.

At least running the unit tests should work, but I realize now that we still need an easy way to launch the debugger with linked libraries.

I'll get started on that soon, it will be a pretty easy change to `dapp`.

**Comment by mbrock** [comment]
I tried `dapp test` with hevm on your repository, and I think it works except that hevm doesn't have the same initial values for the callvalue. I'll fix that in hevm.

**Comment by mbrock** [comment]
Sorry, false alarm, the library support needs a bit more work!

**Comment by mbrock** [comment]
Here we go! The test output using `hevm` is now like this:

```
dapp: Predeploying test library src/exchange/adapter/simpleAdapter.sol:simpleAdapter at d2897d71828e4479bc8a9613004a84dea68837af
dapp: Predeploying test library src/libraries/rewards.sol:rewards at 368d6f5e9394ce286ef65a13b6eaa9374c22e790
dapp: Predeploying test library src/libraries/safeMath.sol:safeMath at b4865f77b58502cd799ade91e858f9ff741ea46e
Running 3 tests for src/libraries/rewards.t.sol:DataFeedTest
...

Running 3 tests for src/assets/Asset.t.sol:AssetTest
...

Running 2 tests for src/participation/Participation.t.sol:ParticipationTest
..

Running 3 tests for src/Fund.t.sol:DataFeedTest
...

Running 2 tests for src/system/Governance.t.sol:DataFeedTest
..

Running 2 tests for src/datafeeds/DataFeed.t.sol:DataFeedTest
..

Running 2 tests for src/version/Version.t.sol:VersionTest
..
```
It should work if you update your `dapp` from Nix, e.g.:

```
$ nix-channel --update
$ nix-env -iA dapp
```

Please let me know if you get an error there about conflicts. If so, you should first do this:

```
$ nix-env -e 'dapp.*'
```

(I corrected the way the `dapp` package is versioned, and I suspect Nix might think a conflict is going on.)

**Comment by travs** [comment]
@mbrock 
Awesome! Thanks for looking into this :)

Looks like it is working on my end; will message if the CI fails for some reason.

For those commands you listed, this happened:

```
$ nix-env -iA dapp                                                                             
error: attribute ‘dapp’ in selection path ‘dapp’ not found 
```

But after doing `nix-env -i dapp` instead, everything seems to work perfectly. Unsure if this is to do with a conflict or not.

**Comment by mbrock** [comment]
Oops, sorry, I meant to write `nix-env -iA dapphub.dapp`, which is an explicit way to specify the package by "path" rather than "name". The name search in Nix works but is really memory hungry so I recommend `-iA`, otherwise people have to enable swap on their VPS just to install!

**Comment by mbrock** [comment]
Oh yeah and there's now also a `dapp debug` command that runs `hevm interactive` but with predeployed libraries. It needs you to specify which compilation output to run, e.g.:

```
$ dapp debug out/libraries/rewards.t.sol.json
```

**Comment by travs** [comment]
This one is fixed now, since 3961b98.
We have also allowed hevm tests to run on other branches without a problem (e.g. current tip of develop: a24f5e7bd24c82f4e514539fdfd9803605b88cff)
